File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.py
s = input()
cro_alpha = ["c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="]
cnt = 0

for i in cro_alpha:
  if i in s:
    cnt += s.count(i)
    s = s.replace(i, " ")

s = s.replace(" ", "")
print(cnt + len(s))

File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.cc
#include <iostream>
#include <string>
using namespace std;

int counter(string str) {
  string cro_alpha[8] = {"c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="};

  for (int i = 0; i < 8; i++) {
    while (1) {
      if (str.find(cro_alpha[i]) != -1) {
        str.replace(str.find(cro_alpha[i]), cro_alpha[i].length(), "#");
      }
      else
        break;
    }
  }

  return str.length();
}

int main() {
  string s;

  cin >> s;
  cout << counter(s);
}